static/js/stream_create.js

```diff
@@ -169,6 +169,13 @@ function create_stream() {
 
     data.stream_post_policy = JSON.stringify(stream_post_policy);
 
+    let message_retention_days = $('#stream_creation_form select[name=stream_message_retention_setting]').val();
```

Comment:
Let's use a better variable name for this temporary value; `message_retention_selection`, perhaps?